website/server/libs/challenges/index.js

```diff
@@ -103,3 +103,29 @@ export function cleanUpTask (task) {
 
   return cleansedTask;
 }
+
+// Create an aggregation query for querying challenges.
+// Ensures that official challenges are listed first.
+export function createChallengeQuery (query) {
```

Comment:
thanks @alecbrick , we're currently having some perf issues with this query so it might take a bit before this is merged. I was wondering if it's possible to get the same result without using the aggregation framework as its purpose is to aggregate multiple docs into a single one and using it like this is a bit outside its scope. But apart from this I like that we're switching from client side sorting to something on the db's side

Reply:
As far as I can tell, there’s no way to sort by whether a challenge has the “is_official” slug when we’re running normal queries on the database. If we brought back the “official” property, that would work. It might be redundant information, but it makes queries easier.

Reply:
@SabreCat @Alys since it's been a while, do you remember what type of perf issues we had here?

Reply:
Sorry, no.